### ISSUE TITLE
ToolShed tool installation error :This repository is not installed correctly

### ISSUE THREAD

**Issue by jvanbraekel** [galaxyproject/galaxy#7059]
Hello,

Wathever toolshed tool I try to install into a local test version of Galaxy 18.05, I get a repository error. 
Full message : "This repository is not installed correctly (see the Repository installation error below). Choose Reset to install from the Repository Actions menu, correct problems if necessary and try installing the repository again."
Sadly, there is no "repository installation error" at all : nothing above,below,aside nor into any logs. 
The only thing present into the logs that may be related to the problem is :

```
galaxy.tools.toolbox.base ERROR 2018-11-29 14:54:03,300 [p:115693,w:0,m:2] [MainThread] Error reading tool configuration file from path 'evolution/codingSnps.xml': [Errno 2] No such file or directory: '/home/galaxy/galaxy/tool-data/codingSnps.loc'
galaxy.tools.toolbox.base ERROR 2018-11-29 14:54:03,301 [p:115693,w:0,m:2] [MainThread] Error reading tool configuration file from path 'evolution/add_scores.xml': [Errno 2] No such file or directory: '/home/galaxy/galaxy/tool-data/add_scores.loc'
galaxy.tools.toolbox.base ERROR 2018-11-29 14:54:03,303 [p:115693,w:0,m:2] [MainThread] Error reading tool configuration file from path 'phenotype_association/sift.xml': [Errno 2] No such file or directory: '/home/galaxy/galaxy/tool-data/sift_db.loc'
```
For instance, with the last version of Fastqc, the installed location should be : /home/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastqc/3e1cdf5406db/fastqc, but the folder home/galaxy/shed_tools remainded empty.   I have checked the permissions and owner : every folder under /home/galaxy is own by galaxy:galaxy and has 775 permissions.  The related config file : "shed_tool_conf.xml" look like : 

```
<?xml version="1.0"?>
<toolbox tool_path="/home/galaxy/shed_tools">
</toolbox>
```

Additionnaly, manual installation of other custom tools trough "tool_conf.xml" work as attented.  

Update : Configuration for a dedicated toolshed sqlite database is used ,  the sqlite file is in the right path and not empty ;  the connection string in galaxy.yaml is :

`install_database_connection: sqlite:////home/galaxy/galaxy/database/toolsheddb.sqlite?isolation_level=IMMEDIATE`

**Comment by jvanbraekel** [comment]
Hello again,

I have try without dedicated toolshed sqlite database : one posgreSQL database for everything,  the issue is still present. Update the instance to the latest galaxy version (18.09) fixed other issues , but this one is still occurring....

Side note :  the database is created and updated, with the help of [manage_db.sh script](https://github.com/galaxyproject/galaxy/blob/dev/manage_db.sh), trough an ansible role derived from [this role](https://github.com/galaxyproject/ansible-galaxy/blob/master/tasks/database.yml).

A screenshot of the error view (with error details missing) :
![2018-12-05 14_19_23-galaxy _ administration](https://user-images.githubusercontent.com/29703119/49516197-597f4d80-f899-11e8-9e35-10fe79e0b869.png)

Fixing the 3  missing '.loc' files was easy , I just had to copy them from the samples present in the 'tool-data' folder, but it was not related to the former problem.

**Comment by jvanbraekel** [comment]
Now that I have a clear error message (see #7080), I solved the issue by using the fix suggested in  [This blog entry](http://galacticengineer.blogspot.com/2018/08/mercurial-based-tool-installation.html).

The repository error was : 'Error cloning repository: [Errno 2] No such file or directory'. Its appenning when you use the uwsgi mules strategy  [uwsgi + mules strategy](https://docs.galaxyproject.org/en/release_18.09/admin/scaling.html) without adding the python virtual env binary path to the python path. Like the blog author, I was missleading by the documentation and tough that the environement line was only needed for handlers strategy. 

So, I fixed the issue by adding this missing line into the uwsgi section of my supervisord.conf file :

`environment = VIRTUAL_ENV="/home/galaxy/galaxy/.venv",PATH="/home/galaxy/galaxy/.venv/bin:%(ENV_PATH)s"`

My curent uwsgi entry in supervisord/conf.d/galaxy;conf looks like :

```
[program:galaxy_uwsgi]
command         = /home/galaxy/galaxy/.venv/bin/uwsgi --yaml /home/galaxy/galaxy/config/galaxy.yml
directory       = /home/galaxy/galaxy
umask           = 022
autostart       = true
autorestart     = true
startsecs       = 10
user            = galaxy
stopsignal      = INT
startretries    = 5
environment = VIRTUAL_ENV="/home/galaxy/galaxy/.venv",PATH="/home/galaxy/galaxy/.venv/bin:%(ENV_PATH)s"
```
A full uwsgi section example like this one, should be inserted in the documentation (into the mule configuration section) to avoid that kind of errors in the future.
